app/scripts/lib/rpc-method-middleware/handlers/switch-ethereum-chain.js

```diff
@@ -103,19 +108,36 @@ async function switchEthereumChainHandler(
       }),
     );
   }
-  const requestData = findExistingNetwork(_chainId, findNetworkConfigurationBy);
-  if (requestData) {
+
+  const requestData = {
+    toNetworkConfiguration: findExistingNetwork(
+      _chainId,
+      findNetworkConfigurationBy,
+    ),
+  };
+
+  requestData.fromNetworkConfiguration = getProviderConfig();
```

Comment:
this will always show the wallet's network as the from network?

Reply:
yeah, at least until we get design update for this confirmation that includes information about the dapps selected network change